docs/docs/daft.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Daft"
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Daft
```

Comment:
so it seems the site can't be actually built when running the docs locally

Reply:
I was able to successfully build the site with `mkdocs serve`!
```
INFO    -  Documentation built in 0.38 seconds
INFO    -  [12:52:52] Watching paths for changes: 'docs', 'mkdocs.yml'
INFO    -  [12:52:52] Serving on http://127.0.0.1:8000/
```

<img width="1180" alt="image" src="https://github.com/apache/iceberg/assets/17691182/54e6353d-dbd6-4392-85dd-3cfe08364db3">


Did you have an error message here that I can look at the debug the issue?

Reply:
I was using https://github.com/apache/iceberg/blob/aff5b39a7dddd22790b6ba47f514860c53e33c00/site/README.md to locally serve the site. @bitsondatadev can you double-check please if the site properly renders when running `./dev/serve.sh` works for you?

Reply:
Yeah, I used to have a "nightly" build in there, but we took it out initially to avoid confusion. I think part of the build can just be to add "local" or something. Currently, the build just grabs the latest semantic version and points latest there, we could just do the same and point /site/docs/docs/local >> /docs and maybe expose another build option to enable that.

Reply:
@bitsondatadev is this PR here ready to go in?